docs/explanation/cryptography.rst

```diff
@@ -0,0 +1,109 @@
+.. _explanation_cryptographic-technology:
+
+Cryptographic technology in Craft Parts
+=======================================
+
+Craft Parts uses cryptographic technologies to obtain and process data from remote
+sources. It does not directly implement its own cryptography, but it does depend
+on external libraries to do so.
+
+Communication with local processes
+----------------------------------
+
+Craft Parts uses the `Requests <https://requests.readthedocs.io/en/latest/>`_ and
+`requests-unixsockets2 <https://gitlab.com/thelabnyc/requests-unixsocket2>`_
+libraries to communicate over Unix sockets with the local `snap daemon (snapd)
+<https://snapcraft.io/docs/installing-snapd>`_. These requests are used to
+fetch information about required snaps. If the snap is missing, Craft
+Parts will install it through snapd. This is done by querying the `snapd
+API <https://snapcraft.io/docs/snapd-api>`_ with URLs built dynamically and
+sanitized by `urllib <https://docs.python.org/3/library/urllib.html>`_.
+
+Overlays
+--------
+
+When :ref:`overlays <overlays>` are enabled, Craft Parts calculates a checksum
+for each part's overlay layer to track when changes are made. The checksums are
+generated using the sha1 algorithm from the `hashlib
+<https://docs.python.org/3/library/hashlib.html>`_ library.
+
+Downloading repositories
+------------------------
+
+For parts that source a git repository, `git <https://git-scm.com/>`_ is used to
+clone the repository. Git uses SSH or HTTPS to communicate with the remote
+repository, depending on the URL provided.
+
+Downloading sources
+-------------------
+
+For parts that source a deb, rpm, snap, tar, zip, or 7z file, the files are
+downloaded using the Requests library.
+
+The integrity of these files can be verified using a
+:ref:`checksum <source_checksum>`. The checksum is verified using hashlib, so all
+`algorithms available
+<https://docs.python.org/3/library/hashlib.html#hashlib.algorithms_available>`_
+to the hashlib library can be used.
+
+Downloading system packages
+---------------------------
+
+System dependencies are downloaded and verified using snapd,
+`apt <https://wiki.debian.org/AptCLI>`_, `dnf <https://dnf.readthedocs.io>`_, and
+`yum <http://yum.baseurl.org>`_.
```

Comment:
We might want to disambiguate between "system packages" here and the earlier mention of debs and rpms. Maybe use an additional heading level to distinguish between sources and dependencies?